src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1673,12 +1676,29 @@ RouterJob(Query *originalQuery, PlannerRestrictionContext *plannerRestrictionCon
  * By default it does not reorder the task list, implying a first-replica strategy.
  */
 static void
-ReorderTaskPlacementsByTaskAssignmentPolicy(Job *job, TaskAssignmentPolicyType
-											taskAssignmentPolicy)
+ReorderTaskPlacementsByTaskAssignmentPolicy(Job *job,
+											TaskAssignmentPolicyType taskAssignmentPolicy,
+											List *placementList)
 {
 	if (taskAssignmentPolicy == TASK_ASSIGNMENT_ROUND_ROBIN)
 	{
-		job->taskList = RoundRobinAssignTaskList(job->taskList);
+		ShardPlacement *primaryPlacement = NULL;
+
+		/*
+		 * We hit a single shard on router plans, and there should be only
+		 * one task in the TaskList
+		 * */
+		Assert(list_length(job->taskList) == 1);
```

Comment:
When built with `--enable-cassert', produces the following warning:

```
warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
```

Reply:
Thanks for pointing this out. I should fine a way to enable this warning flag when I run `make` on my Mac.

I wonder if the CI should fail on this warning.

Reply:
I fixed the order of declarations and statements in this function block. Now there is one new warning on the CI build. It is on a file I did not touch in this PR, and I believe it is safe to say everything is ok.
```
/home/circleci/project/src/backend/distributed/relay/relay_event_utility.c:132:22: warning: implicit declaration of function 'get_relation_constraint_oid' [-Wimplicit-function-declaration]
```